pwndbg/gdblib/memory.py

```diff
@@ -360,3 +360,55 @@ def update_min_addr() -> None:
     global MMAP_MIN_ADDR
     if pwndbg.gdblib.qemu.is_qemu_kernel():
         MMAP_MIN_ADDR = 0
+
+
+def fetch_struct_as_dictionary(
+    struct_name: str,
+    struct_address: int,
+    include_only_fields: list[str] | None = None,
+    exclude_fields: list[str] | None = None,
+):
+    struct_type = gdb.lookup_type("struct " + struct_name)
+    fetched_struct = poi(struct_type, struct_address)
+
+    return pack_struct_into_dictionary(fetched_struct, include_only_fields, exclude_fields)
+
+
+def pack_struct_into_dictionary(
+    fetched_struct: gdb.Value,
+    include_only_fields: list[str] | None = None,
+    exclude_fields: list[str] | None = None,
+):
+    struct_as_dictionary = {}
+
+    if include_only_fields is not None:
```

Comment:
Yeah I see what you're getting at, but I'm keen to leave it as is.
The benefits of a separate loop are, as you mentioned, the efficiency of:
* Only selecting desired fields rather than iterating across every field
* Not trying to parse both include/exclude lists

I haven't seen any giant structs either, but there isn't a strict limit on their size so I'd rather err on the side of caution for the sake of a few duplicated lines here.